src/Model/File.php

```diff
@@ -132,9 +170,101 @@ public function createFromPath(string $path, string $fileName = null): Model
 
         $entity->save();
 
+        // create thumbnail images if needed and possible
+        if ($entity->get('meta_is_image') && $this->createThumbnail === true) {
+            $entity->createThumbnail($path);
+        }
+
         return $entity;
     }
 
+    /**
+     * Create thumbnail images of current image.
+     *
+     * @param string $path Local path to original file
+     *
+     * @return bool True on success
+     */
+    public function createThumbnail(string $path): bool
+    {
+        $this->assertIsEntity();
+
+        $max_width = $this->thumbnailMaxWidth;
+        $max_height = $this->thumbnailMaxHeight;
+
+        $src = imagecreatefromstring(file_get_contents($path));
+        if ($src === false) {
+            return false; // unsupported image type
+        }
+
+        [$width, $height] = getimagesize($path);
+        $x_ratio = $max_width / $width;
+        $y_ratio = $max_height / $height;
+
+        if (($width <= $max_width) && ($height <= $max_height)) {
+            $tn_width = $width;
+            $tn_height = $height;
+        } elseif (($x_ratio * $height) < $max_height) {
+            $tn_height = (int) ceil($x_ratio * $height);
+            $tn_width = $max_width;
+        } else {
+            $tn_width = (int) ceil($y_ratio * $width);
+            $tn_height = $max_height;
+        }
+
+        $tmp = imagecreatetruecolor($tn_width, $tn_height);
+
+        // Check if this image is PNG or GIF, then set if Transparent
+        if ($this->thumbnailFormat === \IMAGETYPE_PNG || $this->thumbnailFormat === \IMAGETYPE_GIF) {
+            imagealphablending($tmp, false);
+            imagesavealpha($tmp, true);
+            $transparent = imagecolorallocatealpha($tmp, 255, 255, 255, 127);
+            imagefilledrectangle($tmp, 0, 0, $tn_width, $tn_height, $transparent);
+        }
+        imagecopyresampled($tmp, $src, 0, 0, 0, 0, $tn_width, $tn_height, $width, $height);
+
+        // create temporary thumb file
+        try {
+            $thumbFile = tmpfile();
+            switch ($this->thumbnailFormat) {
+                case \IMAGETYPE_GIF:
+                    imagegif($tmp, $thumbFile);
+
+                    break;
+                case \IMAGETYPE_JPEG:
+                    imagejpeg($tmp, $thumbFile, 100); // 0=worst quality, 100=best quality
```

Comment:
The defaults are fine here, 75 - 85 is typical, what is the reason for the custom ones?

Reply:
just used max quality for now, but ok - so it should be 80 for jpeg and 6 for png, right?

Reply:
no explicit value :)

Reply:
right, done